demo/elements/print-spec.js

```diff
@@ -1,105 +1,169 @@
 import PresetSpecs from '../preset-specs';
 
+const PresetSpecsNames = Object.keys(PresetSpecs);
+
 class PrintSpec extends HTMLElement {
   constructor() {
     super();
 
     /** @type {boolean} */
-    this.editable_ = false;
+    this.expanded_ = false;
 
     /**
      * Name of the selected preset spec, or null if a custom spec
      * @type {?string}
      */
     this.currentSpecName_ = null;
 
-    /** @type {HTMLButtonElement[]} */
-    this.presetSpecElts = null;
+    /** @type {string} */
+    this.specContent = '';
+
+    /** @type {string} */
+    this.error_ = '';
 
-    /** @type {HTMLTextAreaElement} */
-    this.textElt = null;
+    /** @type {function(boolean):void} */
+    this.onValidityCheckHandler_ = null;
   }
 
   /**
-   * @return {PrintSpec}
+   * @return {?PrintSpec}
    */
   get value() {
-    return JSON.parse(this.textElt.value);
+    try {
+      return JSON.parse(this.specContent);
+    } catch {
+      return null;
+    }
+  }
+
+  onValidityCheck(value) {
+    this.onValidityCheckHandler_ = value;
   }
 
   connectedCallback() {
-    const specNames = Object.keys(PresetSpecs);
+    this.expanded_ = this.getAttribute('expanded') === 'true';
+
+    if (this.getAttribute('select')) {
+      this.selectSpec(this.getAttribute('select'));
+    } else {
+      this.selectSpec(PresetSpecsNames[0]);
+    }
+  }
+
+  selectSpec(name) {
+    this.currentSpecName_ = name;
+    const spec = PresetSpecs[name];
+    this.specContent = JSON.stringify(spec, null, 2);
+    this.refreshDOM();
+  }
+
+  checkSpecValidity() {
+    try {
+      this.error_ = '';
+      JSON.parse(this.specContent);
+    } catch (err) {
+      this.error_ = err.message;
+    }
+
+    const hasError = !!this.error_;
+
+    const jsonValidElt = this.querySelector('.json-validity');
+    if (jsonValidElt) {
+      jsonValidElt.classList.toggle('text-success', !hasError);
+      jsonValidElt.classList.toggle('text-danger', hasError);
+      jsonValidElt.textContent = hasError
+        ? 'The JSON object is invalid: ' + this.error_
+        : 'The JSON object is valid!';
+    }
 
-    this.innerHTML = `
-<div>
-  <div class="dropdown">
-    <button class="btn btn-secondary dropdown-toggle" type="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
-      Choose a predefined spec below
+    if (!!this.onValidityCheckHandler_) {
+      this.onValidityCheckHandler_.call(this, !this.error_);
+    }
+  }
+
+  refreshDOM() {
+    this.innerHTML = this.expanded_
+      ? `
+<div class="border rounded p-1">
```

Comment:
Indentation?

Reply:
Writing HTML in javascript is always weird... prettier does not enforce anything since it's a string. I'm leaving it like that without a better option.